core/LSU.cpp

```diff
@@ -200,61 +236,181 @@ namespace olympia
         ++stores_retired_;
 
         updateIssuePriorityAfterStoreInstRetire_(inst_ptr);
-        uev_issue_inst_.schedule(sparta::Clock::Cycle(0));
+        if(isReadyToIssueInsts_())
+        {
+            ILOG("ROB Ack issue");
+            uev_issue_inst_.schedule(sparta::Clock::Cycle(0));
+        }
 
         ILOG("ROB Ack: Retired store instruction: " << inst_ptr);
     }
 
     // Issue/Re-issue ready instructions in the issue queue
     void LSU::issueInst_()
     {
+        if(allow_speculative_load_exec_ && replay_buffer_.size() > replay_buffer_size_){
```

Comment:
why are we checking if allow_speculative_load_exec is true here? Even if speculative load exec is false, a load or a store can miss in MMU or cache and result in replay and hence, would allocate into the replay buffer during issue.

Reply:
We use the replay buffer only when speculative_load_exec is true. I can remove the check since replay_buffer would default initialize to the size of the load store queue